app/controllers/admin/ccms_queues_controller.rb

```diff
@@ -0,0 +1,44 @@
+module Admin
+  class CCMSQueuesController < AdminBaseController
+    before_action :authenticate_admin_user!
+
+    def index
+      @in_progress = CCMS::Submission.where.not(aasm_state: %w[completed abandoned]).order(created_at: :desc)
+    end
+
+    def show
+      submission
+      legal_aid_application
+    end
+
+    def reset_and_restart
+      execute_submission_method(:restart_from_beginning!)
+    end
+
+    def restart_current_submission
+      execute_submission_method(:restart_current_step!)
+    end
+
+  private
+
+    def submission
+      @submission ||= CCMS::Submission.find(submission_id)
+    end
+
+    def legal_aid_application
+      @legal_aid_application ||= submission.legal_aid_application
+    end
+
+    def submission_id
+      params.require(:id)
+    end
+
+    def execute_submission_method(method)
+      result = submission.send(method)
+      status = result ? :notice : :error
+      message = status.eql?(:error) ? "Restarting submission #{submission_id} failed" : "#{method.to_s.humanize} submission #{submission_id}"
+      flash[status] = message
+      redirect_to action: :show
+    end
```

Comment:
Happy with `public_send` :+1:

slight tweak to your suggestion...
```ruby
if submission.public_send(method)
  redirect_to admin_ccms_queue_url(submission.id), notice: "#{method.to_s.humanize} submission #{submission_id}"
else
  redirect_to admin_ccms_queue_url(submission.id), flash: { error: "Restarting submission #{submission_id} failed" }
 end
```
Apparently `redirect_to action: :name` strips the flash notifications, whereas  using the url_helpers lets them through

Also, only notice and alert have [built-in helpers](https://api.rubyonrails.org/v7.0.4.2/classes/ActionDispatch/Flash/FlashHash.html) so the error has to be in a flash hash setter

Reply:
Are we able to use `_path` helpers instead of `_url`?

Also, if we wanted to we could add this line to `ApplicationController`

```ruby
add_flash_types :error
```

Reply:
Turns out - the error return was optimistic... `process_async!` will always return a value and then fail (and raise errors) elsewhere

Therefore it is currently being removed :+1: